detekt/defs.bzl

```diff
@@ -134,11 +140,46 @@ def _impl(ctx):
         arguments = [java_arguments, detekt_arguments],
     )
 
-    return [DefaultInfo(files = depset(action_outputs))]
+    # Note: this is not compatible with Windows, feel free to submit PR!
+    # text report-contents are always printed to shell
+    final_result = ctx.actions.declare_file(ctx.attr.name + ".sh")
+    ctx.actions.write(
+        output = final_result,
+        content = """
+#!/bin/bash
+set -euo pipefail
+exit_code=$(cat {execution_result})
+cat {text_report}
```

Comment:
Thinking about this more - might actually be better to check that the contents of this file actually context text that needs to be printed to the console before actually printing it so that this test rule doesn't unintentionally dump unnecessary newlines to the console.

Reply:
just to double-check, do you mean we should add a check to verify the contents of the file to ensure it has some content before cat-ing it to the console?

Reply:
modified to:
```
#!/bin/bash
set -euo pipefail
exit_code=$(cat {execution_result})
report=$(cat {text_report})
if [ ! -z "$report" ]; then
    echo "$report"
fi
exit "$exit_code"
```